migrations/updateUserProfileAndMissMatchOfRoleInformation/updateProjectWithProfileData.js

```diff
@@ -0,0 +1,449 @@
+              let userRoleInformationkeys = Object.keys(
+                projectSeggregate.userProfileMissing[count].userRoleInformation
+              );
+              userProfile = _.omit(userProfile, ["userLocations"]);
+              userProfile.userLocations = [];
+              for (let i = 0; i < userRoleInformationkeys.length; i++) {
+                if (userRoleInformationkeys[i].toUpperCase() === "SCHOOL") {
+                  //get school data
+                  let filterData = {
+                    code: projectSeggregate.userProfileMissing[count]
+                      .userRoleInformation[userRoleInformationkeys[i]],
+                  };
+                  let schoolData = await locationSearch(filterData);
+                  if (schoolData.success) {
+                    userProfile.userLocations.push(...schoolData.data);
+                  }
+                } else if (
+                  userRoleInformationkeys[i].toUpperCase() === "ROLE"
+                ) {
+                  //get RoleInformation
+                  let userRoles =
+                    projectSeggregate.userProfileMissing[
+                      count
+                    ].userRoleInformation[userRoleInformationkeys[i]].split(
+                      ","
+                    );
+                  userProfile.profileUserType = new Array();
+                  for (let j = 0; j < userRoles.length; j++) {
+                    if (userRoles[j].toUpperCase() === "TEACHER") {
+                      // If subRole is teacher
+                      userProfile.profileUserType.push({
+                        subType: null,
+                        type: "teacher",
+                      });
+                    } else {
+                      // If subRole is not teacher
+                      userProfile.profileUserType.push({
+                        subType: userRoles[j].toLowerCase(),
+                        type: "administrator",
+                      });
+                    }
+                  }
+                } else {
+                  // get Other RoleInformation data
+                  let filterData = {
+                    id: projectSeggregate.userProfileMissing[count]
+                      .userRoleInformation[userRoleInformationkeys[i]],
+                  };
+                  let userLocations = await locationSearch(filterData);
+                  if (userLocations.success) {
+                    userProfile.userLocations.push(...userLocations.data);
+                  }
+                }
+              }
+            }
+
+            let updateObject = {
+              updateOne: {
+                filter: {
+                  _id: projectIdWithoutUserProfile,
+                },
+                update: {
+                  $set: {
+                    userProfile: userProfile,
+                  },
+                },
+              },
+            };
+
+            updateProjectWithuUserProfile.push(updateObject);
```

Comment:
@ankitpws - Fix spelling mistake

Reply:
fixed updateProjectWithuUserProfile to updateProjectWithUserProfile